src/components/code/index.tsx

```diff
@@ -14,41 +15,202 @@ import styles, {
+            downloadButton={downloadButton}
+            downloadCallback={downloadCallback}
+            wrapLongLines={wrapLongLines}
+            {...props}
+          />
+        )}
+      />
+    );
+
+  // if it's not expandable then render it Normaly
+  return (
+    <CodeSnippet
+      content={content}
+      title={title}
+      language={language}
+      copyButton={copyButton}
+      downloadButton={downloadButton}
+      downloadCallback={downloadCallback}
+      wrapLongLines={wrapLongLines}
+      {...props}
+    />
+  );
+};
+
+export default Code;
+
+// Normal Code Component
+interface CodeSnippetProps
+  extends Omit<CodeProps, 'popupProps' | 'expandable'> {
+  contentToShow: string;
+  maxHeightOfCode?: string;
+}
+const CodeSnippet: FC<CodeSnippetProps> = ({
+  title,
+  content,
+  contentToShow,
+  language,
+  copyButton,
+  downloadButton,
+  downloadCallback,
+  wrapLongLines,
+  showLineNumbers,
+  copyCallback,
+  maxHeightOfCode,
+  ...props
+}) => {
   const theme = useTheme<ITheme>();
 
+  return (
+    <Flex width="100%" sx={{ ...styles }} height="100%">
+      <Flex width="100%" sx={{ ...codeHeaderStyles }}>
+        <Box flexGrow={1} ml="8px" my={1}>
+          {title}
+        </Box>
+        {downloadButton && (
+          <DownloadButton
+            content={content}
+            downloadCallback={downloadCallback}
+            title={title}
+          />
+        )}
+        {copyButton && (
+          <CopyButton content={content} copyCallback={copyCallback} />
+        )}
+      </Flex>
+      <Flex
+        width="100%"
+        variant="code"
+        height={maxHeightOfCode}
+        {...props}
+        p={0}
+      >
+        <SyntaxHighlighter
+          wrapLongLines={wrapLongLines}
+          showLineNumbers={showLineNumbers}
+          lineNumberStyle={{
+            ...lineNumberStyles,
+            background: theme.colors.grayShade1,
+          }}
+          language={language}
+          customStyle={{
+            ...boxStyles,
+            paddingLeft: showLineNumbers ? '0px' : '20px',
+          }}
+        >
+          {contentToShow}
+        </SyntaxHighlighter>
+      </Flex>
+    </Flex>
+  );
+};
+
+// Download Button
+const DownloadButton: FC<
+  Pick<CodeProps, 'content' | 'downloadCallback' | 'title'>
+> = ({ content, downloadCallback, title }) => {
+  const download = () => {
+    if (downloadCallback) {
+      downloadCallback();
+    } else {
+      saveToFile(title || 'download', content);
+    }
+  };
+
+  return (
+    <Box>
+      <Button intent="ghost" sx={{ ...buttonsStyles }} onClick={download}>
+        {icons.download}
+        <Value ml="5px" mt="1px">
+          download
+        </Value>
+      </Button>
+    </Box>
+  );
+};
+
+// Copy Button
+const CopyButton: FC<Pick<CodeProps, 'copyCallback' | 'content'>> = ({
+  copyCallback,
+  content,
+}) => {
+  const [copied, setCopied] = useState(false);
+
   const useCodeKey = async () => {
```

Comment:
I know this is old code too, but it breaks conventions not to use `use-` prefix for anything besides hooks
can you rename this in the PR?

Reply:
good one, sure will update the name